src/cli/index.ts

```diff
@@ -30,8 +31,58 @@ function run(): void {
 function generate(args: { config: string }): void {
   const config = parseConfig(args.config);
 
-  const generator = new CodeGenerator();
-  generator.generate(config);
+  const generator = new CodeGenerator(
```

Comment:
Feels like it's not a good idea to move these heavy logic code here

Reply:
These logic are all CLI related. We do can move them to a dedicated file.